packages/wrangler/CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+});
+
+// Define the command args, implementation and metadata
+const command = defineCommand({
+	command: "wrangler kv key put", // the full command including the namespace
+	metadata: {
+		description: "Put a key-value pair into a Workers KV namespace",
+		status: "stable",
+	},
+	args: {
+		key: {
+			type: "string",
+			description: "The key to put into the KV namespace",
+			required: true,
+		},
+		value: {
+			type: "string",
+			description: "The value to put into the KV namespace",
+			required: true,
+		},
+		"namespace-id": {
+			type: "string",
+			description: "The namespace to put the key-value pair into",
+			required: true,
+		},
+	},
+	// the positionalArgs defines which of the args are positional and in what order
+	positionalArgs: ["key", "value"],
+	handler(args) {
+		// implementation here
+	},
+});
+```
+
+2. global vs shared vs command specific (named + positional) args
+
+- Command-specific args are defined in the `args` field of the command definition. Command handlers receive these as a typed object automatically. To make any of these positional, add the key to the `positionalArgs` array.
+- You can share args between commands by declaring a separate object and spreading it into the `args` field. Feel free to import from another file.
+- Global args are shared across all commands and defined in `src/commands/global-args.ts` (same schema as command-specific args). They are passed to every command handler.
+
+3. (get a type for the args)
+
+You may want to pass your args to other functions. These functions will need to be typed. To get a type of your args, you can use `typeof command.args`.
+
+4. implement the command handler
+
+A command handler is just a function that receives the args as the first param. This is where you will want to do API calls, I/O, logging, etc.
+
+- api calls
+
+Define API response type. Use `fetchResult` to make API calls. `fetchResult` will throw an error if the response is not 2xx.
+
+```ts
+await fetchResult(
+	`/accounts/${accountId}/workers/services/${scriptName}`,
+	{ method: "DELETE" },
+	new URLSearchParams({ force: needsForceDelete.toString() })
+);
+```
+
+- logging
+
+Do not use `console.*` methods to log. You must import and use the `logger` singleton.
+
+- error handling - UserError vs Error
+
+Throw `UserError` for errors _caused_ by the user -- these are not sent to Sentry whereas regular `Error` are and show be used for unexpected exceptions.
+
+Errors are caught at the top-level and formatted for the console.
+
+## Best Practices
+
+### Status / Deprecation
+
+Status can be alpha, private-beta, open-beta, or stable. Breaking changes can freely be made in alpha or private-beta. Try avoid breaking changes in open-beta but are acceptable and should be called out in changeset.
+
+Stable commands should never have breaking changes.
+
+### Changesets
+
+Run `npx changesets` from the top of the repo. New commands warrant a "minor" bump. Please explain the functionality with examples.
```

Comment:
This should link/reference the changesets guidelines in the top-level contributing guide (i.e. `feat:` etc...). It might also be nice to include an example of a good changeset for adding a new command